tests/lib/prepare.sh

```diff
@@ -658,19 +662,18 @@ prepare_ubuntu_core() {
 
 cache_snaps(){
     # Pre-cache snaps so that they can be installed by tests quickly.
-    # This relies on a behavior of snapd where .partial files are
-    # used for resuming downloads.
-    (
-        set -x
-        cd "$TESTSLIB/cache/"
-        # Download each of the snaps we want to pre-cache. Note that `snap download`
-        # a quick no-op if the file is complete.
-        for snap_name in "$@"; do
-            snap download "$snap_name"
-        done
+    # This relies on a behavior of snapd which snaps installed are
+    # cached and then used when need to the installed again
+
+    # Download each of the snaps we want to pre-cache. Note that `snap download`
+    # a quick no-op if the file is complete.
+    for snap_name in "$@"; do
+        snap download "$snap_name"
+
         # Copy all of the snaps back to the spool directory. From there we
         # will reuse them during subsequent `snap install` operations.
-        cp -- *.snap /var/lib/snapd/snaps/
-        set +x
-    )
+        snap_file=$(ls "${snap_name}"_*.snap)
+        mv "${snap_file}" /var/lib/snapd/snaps/"${snap_file}".partial
+        rm -f "${snap_name}"_*
```

Comment:
If this is removing the assert file then please adjust the glob. If this is doing something else can you please add a comment explaining it?

Reply:
done